moVirt/src/main/java/org/ovirt/mobile/movirt/ui/auth/AuthenticatorActivity.java

```diff
@@ -206,16 +207,24 @@ public void finishLogin() {
             setLoginInProgress(false);
         } catch (HttpClientErrorException e) {
             setLoginInProgress(false);
-            switch (e.getStatusCode()) {
-                case NOT_FOUND:
-                    messageHelper.showError(ErrorType.LOGIN, e, getString(R.string.login_error_bad_address_suffix));
-                    break;
-                case UNAUTHORIZED:
-                    messageHelper.showError(ErrorType.LOGIN, e, getString(R.string.login_error_incorrect_username_password));
+            HttpStatus statusCode = e.getStatusCode();
+
+            switch (statusCode.series()) {
+                case REDIRECTION:
+                    messageHelper.showError(ErrorType.USER, e.getMessage());
                     break;
                 default:
-                    messageHelper.showError(ErrorType.LOGIN, messageHelper.createMessage(e));
-                    break;
+                    switch (statusCode) {
```

Comment:
I hope switch in switch makes sense here, because we have different status codes for each series